src/OpenCV/Contrib/Juicy.hs

```diff
@@ -0,0 +1,179 @@
+import Data.Int (Int32)
+
+import Control.Monad (forM_)
+import Control.Monad.Primitive (PrimMonad)
+
+import Linear.V4 (V4)
+
+import OpenCV
+import OpenCV.Unsafe (unsafeRead, unsafeWrite)
+import Codec.Picture.Types 
+
+-- list of pointers at a given byte distance from a base one
+plusPtrS :: Ptr a -> Int -> [Ptr b]
+plusPtrS p n = map (plusPtr p) [0..n-1]
+
+-- multiple peek
+peekS p n = mapM peek (plusPtrS p n)
+
+-- multiple poke
+pokeS p n xs = sequence_ (zipWith poke (plusPtrS p n) xs)
+
+instance Storable (PixelRGB8) where
+    peek p =  peekS p 3 >>= \[b,r,g] -> return (PixelRGB8 r g b)
+    poke p (PixelRGB8 r g b) = pokeS p 3 [b,g,r]
+    sizeOf _ = 3
+    alignment _ = 0
+
+instance Storable (PixelRGB16) where
+    peek p =  peekS p 3 >>= \[b,r,g] -> return (PixelRGB16 r g b)
+    poke p (PixelRGB16 r g b) = pokeS p 3 [b,g,r]
+    sizeOf _ = 3
+    alignment _ = 0
+
+instance Storable (PixelRGBF) where
+    peek p =  peekS p 3 >>= \[b,r,g] -> return (PixelRGBF r g b)
+    poke p (PixelRGBF r g b) = pokeS p 3 [b,g,r]
+    sizeOf _ = 3
+    alignment _ = 0
+
+instance Storable (PixelRGBA8) where
+    peek p =  peekS p 4 >>= \[b,r,g,a] -> return (PixelRGBA8 r g b a)
+    poke p (PixelRGBA8 r g b a) = pokeS p 4 [b,g,r,a]
+    sizeOf _ = 4
+    alignment _ = 0
+
+instance Storable (PixelRGBA16) where
+    peek p =  peekS p 4 >>= \[b,r,g,a] -> return (PixelRGBA16 r g b a)
+    poke p (PixelRGBA16 r g b a) = pokeS p 4 [b,g,r,a]
+    sizeOf _ = 4
+    alignment _ = 0
+
+instance Storable (PixelYA8) where
+    peek p =  peekS p 2 >>= \[b,g] -> return (PixelYA8 b g)
+    poke p (PixelYA8 b g) = pokeS p 2 [b,g]
+    sizeOf _ = 2
+    alignment _ = 0
+
+instance Storable (PixelYA16) where
+    peek p =  peekS p 2 >>= \[b,g] -> return (PixelYA16 b g)
+    poke p (PixelYA16 b g) = pokeS p 2 [b,g]
+    sizeOf _ = 2
+    alignment _ = 0
+
+-- | map Pixel types to a depth
+type family JD a 
+
+-- | map Pixel types to a number of channels
+type family JC a :: Nat 
+
+type instance JD Pixel8 = Word8
+type instance JD Pixel16 = Word16 
+type instance JD PixelF = Float
+type instance JD PixelYA8 = Word8
+type instance JD PixelYA16 = Word16
+type instance JD PixelRGB8 = Word8
+type instance JD PixelRGB16 = Word16
+type instance JD PixelRGBF = Float
+type instance JD PixelRGBA8 = Word8
+type instance JD PixelRGBA16 = Word16
+type instance JD PixelYCbCr8 = Word8
+type instance JD PixelCMYK8 = Word8
+type instance JD PixelCMYK16 = Word16
+
+type instance JC Pixel8 = 1
+type instance JC Pixel16 = 1
+type instance JC PixelF = 1
+type instance JC PixelYA8 = 2
+type instance JC PixelYA16 = 2
+type instance JC PixelRGB8 = 3
+type instance JC PixelRGB16 = 3
+type instance JC PixelRGBF = 3
+type instance JC PixelRGBA8 = 4
+type instance JC PixelRGBA16 = 4
+type instance JC PixelYCbCr8 = 3
+type instance JC PixelCMYK8 = 4
+type instance JC PixelCMYK16 = 4
+
+-- | An OpenCV bidimensional matrix
+type Mat2D h w c d = Mat ('S '[h,w]) c d 
```

Comment:
In other places in this library we write out the `c` and `d` parameters in full like `channels` and `depth`. Please also do that here for consistency.